snooty/postprocess.py

```diff
@@ -710,6 +713,120 @@ def enter_page(self, fileid_stack: FileIdStack, page: Page) -> None:
                 )
 
 
+class GuidesHandler(Handler):
+    """Constructs a dictionary of chapters and their data and returns metadata on individual guides."""
+
+    def __init__(self, context: Context) -> None:
+        super().__init__(context)
+        self.chapters: Dict[str, n.SerializedNode] = {}
```

Comment:
I would *so* much rather have this be a typed object. We overuse the `Dict[str, n.SerializedNode]` idiom as it is. Could you define a dataclass, or better NamedTuple, and turn it into a dictionary only when actually saving it into the metadata collection?

Reply:
Mm, incidentally the `dataclasses.asdict()` function automatically (shallowly) converts a dataclass to a dictionary

Reply:
Updated it to use dataclass and `asdict()`. Let me know if you'd prefer me to make it into a `NamedTuple` instead!

Reply:
Dataclass is great! Thanks!